File: PJDesignBackStageAPI/App.Model/Unit/CreateOrUpdateSetting.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace App.Model
{
    public class CreateOrUpdateSettingRequest
    {
        public int UnitId { get; set; }
        public object? Content { get; set; }
        public byte EditStatus { get; set; }
        public SettingNote? Note { get; set; }
    }

    public class SettingNote
    {
        public string Name { get; set; } = null!;
        public DateTime Date { get; set; }
        public string Note { get; set; } = null!;
    }
}

File: PJDesignBackStageClient/src/app/shared/models/create-or-update-setting.ts
import { EditStatus } from "./enums";
import { ReviewNote } from "./review-note";

export class CreateOrUpdateSettingRequest {
  unitId: number;
  content: object;
  editStatus: EditStatus | undefined;
  note?: ReviewNote;

  constructor(unitId: number, content: object, editStatus: EditStatus | undefined, note?: ReviewNote) {
    this.unitId = unitId;
    this.content = content;
    this.editStatus = editStatus;
    this.note = note;
  }
}

